### PR TITLE
Improve testing section of README

### PR DESCRIPTION
**Pull request by cadejacobson**
When ensuring the changes I made worked, I followed the instructions in the README and saw that I passed all unit tests (only 1). However, I remembered there being more tests than this, so I dug into why this occurred. The README only specifies to run `cargo test`, but does not specify that this will not run every test if you are in the root directory, which may be difficult to understand for new Rust users and/or lead to improper changes being made to the repo. This PR simply adds improved guidance on how to run unit tests.

## How to use

There is no additional usage for this PR.

## Testing done

There is no additional testing done for this PR.

### CODE REVIEW

**Comment by cadejacobson** [comment]
@microsoft-github-policy-service agree

File: README.md
Comment (jeremycline) [inline README.md:49]: Shame on me for not updating this in #96. We could document here that users should run `cargo test --workspace` to pick up all the tests.

Comment (cadejacobson) [reply README.md:49]: Ah, perfect! That's an easy fix.

**Review by jeremycline — APPROVED** [review]
✨

**Comment by dongsupark** [comment]
Nice improvement. Thanks.

One minor note: after merging this PR, the final commit looks like that:
```
Author: Cade Jacobson <91996442+cadejacobson@users.noreply.github.com>
Date:   Tue Nov 26 17:14:43 2024 -0700

    Improve testing section of README (#148)
    
    * Improve testing section of README
    
    * Correct testing line
    
    ---------
    
    Co-authored-by: Cade Jacobson <cade.jacobson@mines.sdsmt.edu>
```

It would be great if the author and the committer would have the same email address, in order not to have another `Co-authored-by` field.
If that is not possible, we should remove unnecessary fields from the commit message, just before merging the PR. That's what the merger should do.

**Pull request merged**
